core/src/autogluon/core/searcher/gp_searcher.py

```diff
@@ -208,6 +209,37 @@ def _to_config_cs(self, config):
         return _to_config_cs(self.gp_searcher.hp_ranges.config_space, config)
 
 
+class ConstrainedGPFIFOSearcher(GPFIFOSearcher):
+    def __init__(self, configspace, **kwargs):
+        _gp_searcher = kwargs.get('_constrained_gp_searcher')
+        if _gp_searcher is None:
+            kwargs['configspace'] = configspace
+            if 'initial_scoring' in kwargs:
+                self.initial_scoring = kwargs.get('initial_scoring', 'acq_func')
+                assert self.initial_scoring == 'acq_func', 'Thompson sampling is not supported for Constrained BO.'
+            else:
+                self.initial_scoring = 'acq_func'
+            _kwargs = check_and_merge_defaults(
+                kwargs, *constrained_gp_fifo_searcher_defaults(),
+                dict_name='search_options')
+            _gp_searcher = constrained_gp_fifo_searcher_factory(**_kwargs)
+        super().__init__(
+            _gp_searcher.hp_ranges.config_space,
+            reward_attribute=kwargs.get('reward_attribute'))
+        self.gp_searcher = _gp_searcher
+        # This lock protects gp_searcher. We are not using self.LOCK, this
+        # can lead to deadlocks when superclass methods are called
+        self._gp_lock = mp.Lock()
+
+    def update(self, config, **kwargs):
+        BaseSearcher.update(self, config, **kwargs)
+        with self._gp_lock:
+            config_cs = self._to_config_cs(config)
+            self.gp_searcher.update(
+                config_cs, reward=kwargs[self._reward_attribute],
+                constraint=kwargs[DEFAULT_CONSTRAINT_METRIC])
```

Comment:
Again, DEFAULT_CONSTRAINT_METRIC hardcoded here. Please make it hardcoded everywhere then, do not let the user choose.

Reply:
Oh wait, no! This name is user facing. You cannot use DEFAULT_CONSTRAINT_METRIC here, you need to use something like self._constraint_attribute !

OK, there are:
- Internal names, in internal searcher code: You can fix them and hardcode
- User facing names, which appear in train_fn. These have to be arguments, of course

Reply:
I've now created a self._constraint_attribute, which is user facing, and hardcoded the rest.